pinot-query-runtime/src/test/resources/queries/FromExpressions.json

```diff
@@ -57,9 +57,14 @@
       },
       {
         "psql": "7.2.1.1",
-        "ignored": true,
+        "comments": "select all doesn't work because h2 output the rows in different order from Pinot and test framework doesn't consider the column order for rows and blindly assume they are in the same order",
         "description": "RIGHT OUTER JOIN",
-        "sql": "SELECT * FROM {tbl1} RIGHT OUTER JOIN {tbl2} ON {tbl1}.col1 = {tbl2}.col1"
+        "sql": "SELECT {tbl1}.num, {tbl1}.name, {tbl2}.num, {tbl2}.value FROM {tbl1} RIGHT OUTER JOIN {tbl2} ON {tbl1}.num = {tbl2}.num"
```

Comment:
could you add tests 
1. with inequality conditions?
2. with mixed column selections such as `{tbl1}.num, {tbl2}.num, {tbl1}.name, {tbl2}.value`
3. with mixed transform functions `{tbl1}.num > {tbl2}.num} + 1`

Reply:
Addressed in the other PR.
Nice test and caught a bug.
Unfortunately,  the inequality join sometimes doesn't work because we don't handle the null value. This is not a regression because left join has the same problem. :(